File: Models/Usuario.cs
using System.ComponentModel.DataAnnotations;

namespace Lavanderia.Models
{
public class Usuario
{
    public int Id { get; set; }

    public string Nombre { get; set; } = string.Empty;

    public string Apellido { get; set; } = string.Empty;

    public string Correo { get; set; } = string.Empty;

    public string Telefono { get; set; } = string.Empty;

    public string Cargo { get; set; } = string.Empty;

    public bool Activo { get; set; } = true;

    public ICollection<Reserva> Reservas { get; set; } = new List<Reserva>();
}
}

File: src/domain/models/Usuario.ts
/**
 * Modelo Usuario (empleado de RefreshProClean)
 * Basado en la tabla SQL "Usuarios" (Id, Nombre, Apellido, Correo, Telefono, Cargo, Activo).
 * Se usa solo como estructura en memoria, no hay conexión a base de datos.
 */

export interface Usuario {
  id: string;
  nombre: string;
  apellido: string;
  correo: string;
  telefono: string;
  cargo: string; // Ej: "Operario", "Recepcionista", "Repartidor"
  activo: boolean;
}

export interface CrearUsuarioParams {
  id: string;
  nombre: string;
  apellido: string;
  correo: string;
  telefono: string;
  cargo: string;
  activo?: boolean;
}

export const crearUsuario = ({
  id,
  nombre,
  apellido,
  correo,
  telefono,
  cargo,
  activo = true,
}: CrearUsuarioParams): Usuario => ({
  id,
  nombre,
  apellido,
  correo,
  telefono,
  cargo,
  activo,
});
